sw/device/tests/sysrst_ctrl_in_irq_test.c

```diff
@@ -213,22 +210,42 @@ void sysrst_ctrl_key_combo_detect(dif_sysrst_ctrl_key_combo_t key_combo,
  * External interrupt handler.
  */
 void ottf_external_isr(uint32_t *exc_info) {
-  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kPlicTarget, &irq_id));
-
-  peripheral = (top_earlgrey_plic_peripheral_t)
-      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
-
-  if (peripheral == kTopEarlgreyPlicPeripheralSysrstCtrlAon) {
-    irq =
-        (dif_sysrst_ctrl_irq_t)(irq_id -
-                                (dif_rv_plic_irq_id_t)
-                                    kTopEarlgreyPlicIrqIdSysrstCtrlAonEventDetected);
-    CHECK_DIF_OK(dif_sysrst_ctrl_irq_acknowledge(&sysrst_ctrl, irq));
+  const uint32_t kPlicTarget = kTopEarlgreyPlicTargetIbex0;
+  dif_rv_plic_irq_id_t plic_irq_id = 0;
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kPlicTarget, &plic_irq_id));
+
+  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[plic_irq_id];
+
+  switch (peripheral) {
+    case kTopEarlgreyPlicPeripheralUart0:
+      if (!ottf_console_flow_control_isr(exc_info)) {
```

Comment:
I wonder how many tests have flow control enabled, yet don't properly handle the interrupt...

Reply:
That's a good catch, I probably modified this sysrst_ctrl test for sival and enabled control flow without realizing that it required the ISR... It's also kind of nasty that every test has to call the ottf console flow isr manually..

Reply:
> It's also kind of nasty that every test has to call the ottf console flow isr manually..

Probably, we would want to have OTTF provide the top-level ISR, and it could insert handlers for its enabled features, then call into a hook provided by the downstream test. If I have missed something, that might risk interfering with how some tests might want to work, though.